ambry-frontend/src/main/java/com/github/ambry/frontend/PostBlobHandler.java

```diff
@@ -339,9 +344,12 @@ private void checkUploadRequirements(BlobProperties blobProperties) throws RestS
         RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.SESSION, true);
         // validate that a max chunk size is set.
         RestUtils.getLongHeader(restRequest.getArgs(), RestUtils.Headers.MAX_UPLOAD_SIZE, true);
+        Long chunkUploadTtl =
+            RestUtils.getLongHeader(restRequest.getArgs(), RestUtils.Headers.CHUNK_UPLOAD_TTL, false);
         // validate that the TTL for the chunk is set correctly.
         long chunkTtl = blobProperties.getTimeToLiveInSeconds();
-        if (chunkTtl <= 0 || chunkTtl > frontendConfig.chunkUploadInitialChunkTtlSecs) {
```

Comment:
If you make the change to the url signing service impls I described, I think you can just change this part of the code to see if chunkTtl <= frontendConfig.chunkUploadMaxChunkTtlSecs and get rid of the less than initialTtl clause.

Reply:
Like Casey suggested, I was also wondering if making the change in url signing service impls is a good idea since chunk uploads seems to originate only with signed URL.

Also, just a question. Apart from extending the maximum TTL of chunk upload to 90 days, are we also trying to provide option for user to specify the TTL value (between 0 to 90 days). 

1. If yes, based on this [code](https://github.com/linkedin/ambry/blob/97623533690e6e0986265f395162dfe224d2862f/ambry-frontend/src/main/java/com/github/ambry/frontend/AmbryUrlSigningService.java#L142) , it looks like we are currently always setting the TTL value in the signed URL to the config value `frontendConfig.chunkUploadInitialChunkTtlSecs` (with a default value of 28 days). Would it require modifying the code there too to instead take user provided value (provided via in x-ambry-ttl header) and set it in the signed URL?
2. If not, it looks like if we modify the config` frontendConfig.chunkUploadInitialChunkTtlSecs` to 90 days, it might satisfy user requirement. But, it might increase our storage space temporarily until the chunks are stitched and made permanent?

Also another question - do we need to validate the value of TTL in `checkUploadRequirements` for chunked requests since this TTL value is coming from signed URL and might already be checked during generation of the URL?

Reply:
Sorry, I was looking at OSS code above. In AmbryLI code, looks like we choose user TTL during generation of signed URLs for chunked uploads. Then just wondering that if we increase the config value `frontendConfig.chunkUploadInitialChunkTtlSecs` to 90 days, would it be sufficient to handle the requirement?

Reply:
> If you make the change to the url signing service impls I described, I think you can just change this part of the code to see if chunkTtl <= frontendConfig.chunkUploadMaxChunkTtlSecs and get rid of the less than initialTtl clause.
I was thinking that constraint our client to use 28days ttl by default and if they have special cases, we can ask them to provide extra header if they are sure it won't affect GDPR compliance. If I understand your ideal correctly, seems that we won't gate them to have ttl larger than 28 days by default, then could we just increase the chunkUploadInitialChunkTtlSecs to 90 days?

Inside checkUploadRequirements, we only set the restriction when request isChunkUpload, are you suggesting instead of set the constraint at the upload blob stage, we should set it up at the getSignedUrl stage?

Reply:
After discuss offline, we will use chunkUploadInitialChunkTtlSecs as the default ttl for chunkupload and allow user to specify longer ttl < 90 when they set x-ambry-ttl header.